### DISCUSSION TITLE
Exposing readonly information about the VS Code UI

### DISCUSSION THREAD

**Discussion by joaomendoncaa** [microsoft/vscode-discussions#610] [Extension Development QnA] [answered]
I'm trying to develop a small extension that needs to calculate the dimensions of the currently active text editor. And I've came to the conclusion that it is a huge hassle to be able to do it without making the user modify source files and so on... I don't understand how this is such a huge challenge and it would be solved if the UI was exposed in a "readonly" approach. 

People say that accessing the DOM is intentionally not available for security reasons (I want to believe performance and security are not the real reasons behind it) but seems to be this is blocking a lot of cool niche extensions to flourish. 

Let's open an already awesome ecosystem to even more integrations!

**Comment by alefragnani** [comment, marked as answer]
Hi @joaomendoncaa ,

The restrictions and their reasons are described here (https://code.visualstudio.com/api/extension-capabilities/overview#restrictions), and as you can see, not only performance but also stability and breaking changes are key factors on the decision.

I didn’t understand exactly what you need, and what is missing in the current API, but based on your description, it seems you need _read_ information, not _write_. If that’s the case, that “restrictions” info may not be entirely applied.

So, I guess the best approach for you is to provide more details about what you are trying to achieve, how far you could go with the current APIs, and what you are missing. With that, if new APIs are really necessary, you could open a feature request in VS Code repo, so the team could take a look and add to their backlog, if applicable.

Hope this helps.

**Reply by joaomendoncaa** [reply]
I wanted to make a really small extension that added into the split functionality where, depending on the aspect ratio available on the current text editor the split would be made vertical or horizontal accordingly. 
**i.e: If you split and the current text editor width is bigger than it's height it would split horizontally, otherwise it would split vertically.**
I got flabbergasted on how something like this is *impossible* with the current APIs and tried to reach out and see if I was missing something here.

I'll try to make a feature request but honestly I might not be the first one to request a readonly API for the UI with no success for years?

Anyways, thanks a lot for the feedback!
